erroranalysis/erroranalysis/analyzer/error_analyzer.py

```diff
@@ -476,6 +476,8 @@ def compute_importances(self, error_correlation_method=MUTUAL_INFO):
                                                      IMPORTANCES_THRESHOLD)
             input_data = input_data[indexes]
             diff = diff[indexes]
+        # For numerical data types to make calculation/imputation error-free
+        input_data = input_data.astype(float)
```

Comment:
Looks like this converts all column types to float type. What about data types like integer and strings?

Reply:
`input_data` encodes all values into numericals it seems:

Example with the metadata for the fridge dataset, `input_data` is:

[[104.43172731850086 0.0 1 0.0 2 244 0.0 31385.0 3024 4032 1 4.0]
 [100.46510438294005 0.0 1 0.0 2 244 0.0 32625.0 3024 4032 1 3.0]
  [102.94642036826406 0.0 1 0.0 2 244 0.0 32776.0 3024 4032 1 0.0]
  [103.5999867603074 0.0 1 0.0 2 244 0.0 31538.0 3024 4032 1 2.0]
  [99.57597176334652 0.0 1 0.0 2 232 0.0 nan 3024 4032 1 8.0]
  [99.52725370661243 0.0 1 0.0 2 244 0.0 31521.0 3024 4032 1 5.0]
  [105.19050714040694 0.0 1 0.0 2 244 0.0 31914.0 3024 4032 1 6.0]
  [99.96797699503111 0.0 1 0.0 2 232 0.0 nan 3024 4032 1 9.0]
  [105.55769296350459 0.0 1 0.0 2 244 0.0 30998.0 3024 4032 1 7.0]
  [109.44899007223657 0.0 1 0.0 2 244 0.0 26287.0 3024 4032 1 1.0]]
  
  and the nans are currently imputed with zeros as per existing logic. The code change just ensures the existing imputation is successful and avoids the uncaught exception.